src/Thinktecture.Blazor.PwaUpdate/README.md

```diff
@@ -0,0 +1,127 @@
+# Thinktecture.Blazor.PwaUpdate
+
+
+[![NuGet Downloads (official NuGet)](https://img.shields.io/nuget/dt/Thinktecture.Blazor.PwaUpdate?label=NuGet%20Downloads)](https://www.nuget.org/packages/Thinktecture.Blazor.PwaUpdate/)
+
+## Introduction
+
+A Blazor wrapper for the [Service Worker Update](https://web.dev/service-worker-lifecycle/#update-on-reload).
+
+The package allows subscribing to an event that fires as soon as a new update is available.
+
+## Getting started
+
+### Prerequisites
+
+You need .NET 7.0 or newer to use this library.
+
+[Download .NET 7](https://dotnet.microsoft.com/download/dotnet/7.0)
+
+### Platform support
+
+[Platform support for Service Worker](https://caniuse.com/serviceworkers)
+
+### Installation
+
+You can install the package via NuGet with the Package Manager in your IDE or alternatively using the command line:
+
+```
+dotnet add package Thinktecture.Blazor.PwaUpdate
+```
+
+## Usage
+
+The package can be used in Blazor WebAssembly projects.
+
+### Add to service collection
+
+To make the UpdateService available on all pages, register it at the IServiceCollection in `Program.cs` before the host is built:
+
+```csharp
+builder.Services.AddUpdateService();
+
+```
+### Add to Imports
+
+To use the default UpdateModal component on the hole app razor files, register it in the `_Imports.razor` file.
+
+```html
+@using Thinktecture.Blazor.PwaUpdate
+```
+
+### Adapt Publish ServiceWorker
+
+To get to actual service worker use the `skipWaiting`-method in the service worker. This method forces the waiting service worker to become the active service worker.

```

Comment:
```suggestion
By default, service workers will only update when all tabs of the controlled websites have been closed.
To be able to update an application on reload, you need to call the `skipWaiting()` method of the service worker upon installation.
This skips the service worker's waiting phase, and the new version becomes active instantly.
A reload then leads to the new version of the application the be shown.
Please be aware of the potential implications this may have:
If the application relies on a certain version of the service worker or its cache while it is running, this may lead to errors during runtime.
```